File: source_funcs/go_transcoder/MINIMUM_TIME_TO_FINISH_TASKS_WITHOUT_SKIPPING_TWO_CONSECUTIVE/MINIMUM_TIME_TO_FINISH_TASKS_WITHOUT_SKIPPING_TWO_CONSECUTIVE.go
package main

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}

func f_gold(arr []int, n int) int {
	if n <= 0 {
		return 0
	}
	var incl int = arr[0]
	var excl int = 0
	for i := int(1); i < n; i++ {
		var (
			incl_new int = arr[i] + min(excl, incl)
			excl_new int = incl
		)
		incl = incl_new
		excl = excl_new
	}
	return min(incl, excl)
}

File: rust_funcs/cpp_transcoder/MINIMUM_TIME_TO_FINISH_TASKS_WITHOUT_SKIPPING_TWO_CONSECUTIVE/MINIMUM_TIME_TO_FINISH_TASKS_WITHOUT_SKIPPING_TWO_CONSECUTIVE.rs
fn min(x: i32, y: i32) -> i32 {if x < y { x } else { y } }
fn max(x: i32, y: i32) -> i32 {if x > y { x } else { y }}
fn cmpfunc(a: &i32, b: &i32) -> std::cmp::Ordering {a.cmp(b)}
fn len(arr: &[i32]) -> usize {arr.len()}
fn sort(arr: &mut [i32]) {arr.sort_by(cmpfunc);}

use std::f32;

fn f_gold(arr: [f32;2], n: i32) -> i32 {
    if n <= 0 {
        return 0.0;
    }
    let mut incl = arr[0];
    let mut excl = 0.0;
    for i in 1..n as usize {
        let incl_new = arr[i] + incl.min(excl);
        let excl_new = incl;
        incl = incl_new;
        excl = excl_new;
    }
    incl.min(excl)
}
